tap_dynamodb/deserialize.py

```diff
@@ -58,23 +58,61 @@ def _apply_projection(self, record, breadcrumb, output):
                 breadcrumb_key = breadcrumb[0].split('[')[0]
                 index = int(breadcrumb[0].split('[')[1].split(']')[0])
                 if output.get(breadcrumb_key):
-                    output[breadcrumb_key].append(record[breadcrumb_key][index])
+                    # record: {'Artist': 'No One You Know5', 'metadata': ['test1']}
+                    # main breadcrumb = [['metadata[0]'], ['metadata[1]']]
+                    # current breadcrumb = ['metadata[1]']
+                    # current output = {'metadata': ['test1']}
+                    # expected output = {'metadata': ['test1']}
+                    #       as "metadata" has only 1 item and the current breadcrumb is expecting 2nd item
+
+                    # only prepare output if the list field contains data at that index position in record
+                    if len(record.get(breadcrumb_key)) > index:
+                        output[breadcrumb_key].append(record[breadcrumb_key][index])
                 else:
-                    output[breadcrumb_key] = [record[breadcrumb_key][index]]
+                    # record: {'Artist': 'No One You Know5'}
+                    # main breadcrumb = [['metadata[0]']]
+                    # current breadcrumb = ['metadata[0]']
+                    # current output = {}
+                    # expected output = {'metadata': []}
+                    #       as "metadata" does not have any items and the current breadcrumb is expecting 1st item
 
+                    output[breadcrumb_key] = []
+                    # only prepare output if the list field contains data at that index position in record
+                    if record.get(breadcrumb_key) and len(record.get(breadcrumb_key)) > index:
+                        output[breadcrumb_key].append(record[breadcrumb_key][index])
             else:
                 output[breadcrumb[0]] = record.get(breadcrumb[0])
         else:
             if '[' in breadcrumb[0]:
                 breadcrumb_key = breadcrumb[0].split('[')[0]
                 index = int(breadcrumb[0].split('[')[1].split(']')[0])
-                if output.get(breadcrumb_key) is None:
+                if not output.get(breadcrumb_key):
```

Comment:
I think this should be 
```suggestion
                if breadcrumb_key not in output:
```

Reply:
Updated.

Reply:
@cosimon 
As seen in the figure, the particular code was kept to handle such scenarios where the dictionary contains an empty list which would return false if w change it to `if breadcrumb_key not in output:`
Hence, not updated the code.
![image](https://user-images.githubusercontent.com/75604662/149314267-d1c59af4-a9b4-4e75-ab7f-55f5aa1b3a91.png)